### PR TITLE
docs: add cosign keyless documentation

### PR DESCRIPTION
**Pull request by akashsinghal**
Adds documentation for the new cosign keyless experience for trust policy

- Moves docs for previous keyless support to be under legacy section
- Creates a new keyless section explaining how keyless verification is configured
- updates Kubernetes config section to add new fields

### CODE REVIEW

File: docs/plugins/verifier/cosign.md [hunk 1]
```diff
@@ -385,14 +403,19 @@ spec:
...
             }
         ]
     }
 }
 ```
 
-### Usage
+Please note that the `key` is not specified in the config. This is because the keyless verification uses ephemeral keys and certificates, which are signed automatically by the [fulcio](https://github.com/sigstore/fulcio) root CA. Signatures are stored in the [Rekor](https://github.com/sigstore/rekor) transparency log, which automatically provides an attestation as to when the signature was created.
+
+The `rekorURL` MUST be provided for keyless verification. Otherwise, signature validation will fail.
```
Comment (yizha1) [inline docs/plugins/verifier/cosign.md:415]: Why `rekorURL` is a MUST? If this parameter does not exist, the default Sigstore URL should be used, similiar to `cosign verify` command.

File: docs/plugins/verifier/cosign.md [hunk 2]
```diff
@@ -192,16 +208,52 @@ kubectl run demo2 -n default --image=<insert test namespace image reference>
...
+  name: verifier-cosign
+spec:
+  artifactTypes: application/vnd.dev.cosign.artifact.sig.v1+json
+  name: cosign
+  parameters:
+    trustPolicies: # OPTIONAL: [list], trust policies matching keys to scopes
+    - name: # REQUIRED: [string], trust policy name. MUST be unique across policies
+      version: # OPTIONAL: [string], trust policy schema version  
+      scopes: # REQUIRED: [list], string list of scopes
+      tLogVerify: # OPTIONAL: [boolean] enables/disables transparency log verification. default is 'true'
```
Comment (yizha1) [inline docs/plugins/verifier/cosign.md:228]: missing OPTIONAL parameter `rekorURL`?

**Review by yizha1 — COMMENTED** [review]
Thanks @akashsinghal Left some comments.

File: docs/plugins/verifier/cosign.md [hunk 3]
```diff
@@ -44,7 +45,7 @@ A configuration flag `cosignEnabled` is introduced to the ORAS Store configurati
```
[the remark this answers is not in the document]
Comment (akashsinghal) [reply docs/plugins/verifier/cosign.md:45]: Yes this is accurate. The `cosignEnabled` flag is to enable special fetching behavior of `.sig` tag convention of non OCI 1.1 signatures. If you have a OCI 1.1 cosign sig, that would be discovered via referrers list which is done by default for all verifiers and cosign would be invoked accordingly then on those OCI 1.1 signatures. Are you seeing behavior that doesn't not match this?

File: docs/plugins/verifier/cosign.md (hunk 1 above)
Comment (akashsinghal) [reply docs/plugins/verifier/cosign.md:415]: Yeah great point. It ideally shouldn't be and that's why we introduced extra flag `tlogverify` to the new trust policy implementation. This is documenting the current legacy behavior which we will maintain for backwards compatability. This content in the legacy section is just rearranged which is why it's showing as a change.

File: docs/plugins/verifier/cosign.md [hunk 4]
```diff
@@ -102,9 +112,15 @@ The `file` field defines an absolute file path to a local public key. This field
...
 
-### Limitations
+### Keyless
+
+A trust policy can be configued for keyless verification for all images matching the specified `scopes`. Keyless verification can NOT be configured if `keys` are already defined.
+
+Keyless verification supports:
 
-Currently, Cosign trust policies only support key-based configurations. Keyless support will be added soon.
+- certificate transparency log verification: The `ctLogVerify` configuration enables/disables verification of the existence Secure Certificate Timestamp (SCT) entry in the certificate transparency log. By defaault, this is `true`.
```
Comment (junczhu) [inline docs/plugins/verifier/cosign.md:121]: There was a typo `defaault`

Comment (akashsinghal) [reply docs/plugins/verifier/cosign.md]: good catch! thanks

File: docs/plugins/verifier/cosign.md [hunk 5]
```diff
@@ -192,16 +208,53 @@ kubectl run demo2 -n default --image=<insert test namespace image reference>
...
+spec:
+  artifactTypes: application/vnd.dev.cosign.artifact.sig.v1+json
+  name: cosign
+  parameters:
+    trustPolicies: # OPTIONAL: [list], trust policies matching keys to scopes
+    - name: # REQUIRED: [string], trust policy name. MUST be unique across policies
+      version: # OPTIONAL: [string], trust policy schema version  
+      scopes: # REQUIRED: [list], string list of scopes
+      tLogVerify: # OPTIONAL: [boolean] enables/disables transparency log verification. default is 'true'
+      rekorURL: # OPTIONAL: [string] specifies a rekor URL pointing to a transparency log server. default is https://rekor.sigstore.dev
```
Comment (junczhu) [inline docs/plugins/verifier/cosign.md:229]: nit: Just wondering if `Rekor` is better

File: docs/plugins/verifier/cosign.md [hunk 6]
```diff
@@ -192,16 +208,53 @@ kubectl run demo2 -n default --image=<insert test namespace image reference>
...
+          name: # OPTIONAL: [string], name of key stored in referenced provider
+          version: # OPTIONAL: [string], version of named key
+      keyless: # OPTIONAL: keyless verification configuration. Either 'keys' or 'keyless' must be defined
+        ctLogVerify: # OPTIONAL: [boolean] enables/disables certificate transparency log verification. default is 'true'
+        certificateIdentity: # OPTIONAL: [string] exact string identity associated with public certificate
+        certificateIdentityRegExp: # OPTIONAL: [string] string regular expression of matching identity associated with public certificate.
+        certificateOIDCIssuer: # OPTIONAL: [string] exact string OIDC issuer associated with public certificate
+        certificateOIDCIssuerRegExp: # OPTIONAL: [string] string regular expression of matching OIDC issuer associated with public certificate.
+    key: # DEPRECATED,OPTIONAL: [string], absolute file path to public key
+    rekorURL: # DEPRECATED,OPTIONAL: [string], rekor server URL
```
Comment (junczhu) [inline docs/plugins/verifier/cosign.md:242]: nit: Same here

File: docs/plugins/verifier/cosign.md (hunk 5 above)
Comment (junczhu) [reply docs/plugins/verifier/cosign.md:229]: 🤣 nvm, showing inconsistency in their own repo.

Ref: https://github.com/search?q=repo%3Asigstore%2Fdocs+rekor+server&type=code#:~:text=Cosign%20defaults%20to,url%60%20flag

**Comment by akashsinghal** [comment]
TODO: add section showing what the verifier report looks like for keyed and keyless cosign verification

File: docs/plugins/verifier/cosign.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (yizha1) [reply docs/plugins/verifier/cosign.md:45]: Got it. Thanks.

File: docs/plugins/verifier/cosign.md (hunk 1 above)
Comment (yizha1) [reply docs/plugins/verifier/cosign.md:416]: OK, this is under legacy section. Then it makes sense now.

**Review by yizha1 — APPROVED** [review]
LGTM, but I am not a maintainer.

**Review by susanshi — APPROVED** [review]
LGTM thanks Akash!

**Pull request merged**